packages/aws-cdk-lib/aws-appsync/lib/appsync-function.ts

```diff
@@ -152,6 +159,10 @@ export class AppsyncFunction extends Resource implements IAppsyncFunction {
       throw new Error('Mapping templates cannot be used alongside code');
     }
 
+    if (!(props.dataSource instanceof LambdaDataSource) && props.maxBatchSize) {
```

Comment:
```suggestion
    if (props.maxBatchSize && !(props.dataSource instanceof LambdaDataSource)) {
```
A bit clearer to read in my opinion